modules/spotxBidAdapter.js

```diff
@@ -0,0 +1,330 @@
+    if (window.document.getElementById(videoSlotDivId) === null) {
+      utils.logMessage(BIDDER_CODE + ': video_slot HTML node id does not exist on the page');
+      return false;
+    }
+
+    const slotDivId = utils.getBidIdParameter('slot', bid.params.video);
+    if (window.document.getElementById(slotDivId) === null) {
+      utils.logMessage(BIDDER_CODE + ': slot HTML node id does not exist on the page');
+      return false;
+    }
+
+    return true;
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   * from Prebid.js: buildRequests - Takes an array of valid bid requests, all of which are guaranteed to have passed the isBidRequestValid() test.
+   *
+   * @param {BidRequest[]} bidRequests A non-empty list of bid requests which should be sent to the Server.
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function(bidRequests, bidderRequest) {
+    const page = bidderRequest.refererInfo.referer;
+    const isPageSecure = !!page.match(/^https:/)
+
+    const siteId = '';
+    const bid = bidderRequest.bids[0];
+    const channelId = bid.params.video.channel_id;
+    let pubcid = null;
+
+    const videoSlotDiv = window.document.getElementById(utils.getBidIdParameter('video_slot', bid.params.video));
+    const hasVideoElement = videoSlotDiv.querySelectorAll('video').length;
+    const contentWidth = utils.getBidIdParameter('content_width', bid.params.video) ||
+      (hasVideoElement
+        ? videoSlotDiv.querySelectorAll('video')[0].offsetWidth
+        : videoSlotDiv.offsetWidth);
+    const contentHeight = utils.getBidIdParameter('content_height', bid.params.video) ||
+      (hasVideoElement
+        ? videoSlotDiv.querySelectorAll('video')[0].offsetHeight
+        : videoSlotDiv.offsetHeight);
+
+    const spotxImps = bidRequests.map(function(bid) {
+      const secure = isPageSecure || (utils.getBidIdParameter('secure', bid.params) ? 1 : 0);
+
+      const ext = {
+        player_width: contentWidth,
+        player_height: contentHeight,
+        sdk_name: 'Prebid 1+',
+        ad_mute: +!!utils.getBidIdParameter('ad_mute', bid.params.video),
+        hide_skin: +!!utils.getBidIdParameter('hide_skin', bid.params.video),
+        content_page_url: page,
+        versionOrtb: ORTB_VERSION,
+        bidId: bid.bidId,
+        videoSlot: bid.params.video.video_slot
+      };
+
+      if (utils.getBidIdParameter('ad_volume', bid.params.video) != '') {
+        ext.ad_volume = utils.getBidIdParameter('ad_volume', bid.params.video);
+      }
+
+      if (utils.getBidIdParameter('ad_unit', bid.params.video) != '') {
+        ext.ad_unit = utils.getBidIdParameter('ad_unit', bid.params.video);
+      }
+
+      if (utils.getBidIdParameter('outstreamFunction', bid.params.video) != '') {
+        ext.outstreamFunction = utils.getBidIdParameter('outstreamFunction', bid.params.video);
+      }
+
+      if (utils.getBidIdParameter('custom', bid.params.video) != '') {
```

Comment:
Missing documentation. I assume these are the key-values a publisher can specify?

Reply:
You are right